tpl/tplimpl/embedded/templates/partials/_funcs/get-page-images.html

```diff
@@ -0,0 +1,27 @@
+{{- $imgs := slice }}
+{{- $page := . }}
+{{/* Parse page's images parameter. */}}
+{{- range .Params.images }}
+  {{- $img := partial "_funcs/parse-page-image" (dict "Resources" $page.Resources "Image" .) }}
+  {{- $imgs = $imgs | append $img }}
+{{- end }}
+{{/* Find featured image resources if the images parameter is empty. */}}
+{{- if not $imgs }}
+  {{- $images := $.Resources.ByType "image" -}}
+  {{- $featured := $images.GetMatch "*feature*" -}}
+  {{- if not $featured }}{{ $featured = $images.GetMatch "{*cover*,*thumbnail*}" }}{{ end -}}
+  {{- with $featured }}
+    {{- $imgs = $imgs | append (dict
+      "Image" .Name
```

Comment:
Image must either be a image object (with .Width, .Height etc.) or nothing at all (an empty string is fine and practical.

The idea about this is that people could use this partial in their own template and say:

```handlebars
{{ $i := partial "_funcs/get-page-images.html" }}
{{ with $i }}
  {{ with $i.Image }}
      // resize, apply image filters etc.
  {{ else }}
   // just use the .RelPermalink
  {{ end }}
{{ end }}
```